### PR TITLE
i.atcorr: fix out of bound access in trunca, os and iso

### PR DESCRIPTION
**Pull request by ymdatta**
In `trunca()`, under 2 if conditionals (`if rumu[1] > 0.8 and rmu[1] > 0.94`), k and kk values are set to `-1` respectively. When these values are used to access an array, it leads to out of bound access, which is undefined behavior in c++.

Similarly in `os()` and `iso()`, if `taup < h[i]` for all values, iplane would still be `-1` and we would be accessing an array with this as an index, which is undefined behavior.

In both cases, to avoid that, set that index to zero.

This was found using cppcheck tool.

### CODE REVIEW

File: imagery/i.atcorr/computations.cpp [hunk 1]
```diff
@@ -444,7 +444,7 @@ void os(const double tamoy, const double trmoy, const double pizmoy,
     if (ntp == (snt - 1)) {
         /* compute position of the plane layer */
         double taup = tap + trp;
-        iplane = -1;
+        iplane = 0;
         for (int i = 0; i <= ntp; i++)
             if (taup >= h[i])
                 iplane = i;
```
Comment (metzm) [inline imagery/i.atcorr/computations.cpp:450]:
```suggestion
        iplane = -1;
        for (int i = 0; i <= ntp; i++) {
            if (taup >= h[i])
                iplane = i;
        }
        if (iplane == -1) {
            G_fatal_error(_("Position of the plane layer could not be determined"));
        }
```

File: imagery/i.atcorr/computations.cpp [hunk 2]
```diff
@@ -1005,7 +1005,7 @@ void iso(const double tamoy, const double trmoy, const double pizmoy,
...
```
Comment (metzm) [inline imagery/i.atcorr/computations.cpp:1011]:
```suggestion
        iplane = -1;
        for (int i = 0; i <= ntp; i++) {
            if (taup >= h[i])
                iplane = i;
        }
        if (iplane == -1) {
            G_fatal_error(_("Position of the plane layer could not be determined"));
        }
```

**Review by metzm — APPROVED** [review]

File: imagery/i.atcorr/computations.cpp (hunk 1 above)
Comment (ymdatta) [reply imagery/i.atcorr/computations.cpp]: Made some very minor changes to accommodate clang-format in a later commit.

File: imagery/i.atcorr/computations.cpp (hunk 2 above)
Comment (ymdatta) [reply imagery/i.atcorr/computations.cpp]: Again, made some very minor changes to accommodate clang-format in a later commit.

**Comment by echoix** [comment]
I'll proxy the approval once it passes later tonight

**Review by echoix — APPROVED** [review]

**Pull request merged**